### PR TITLE
Posts

### LINKED ISSUE #10: A user can both like and comment on posts created by himself or other users
User Story:
User can add a like or comment to a post which are displayed under the post itself and can be viewed anytime.

### LINKED ISSUE #11: A user can create an account to better organize posts and to easily follow other users
User Story:
User can create an account where postings can be easily organized. To add, having a unique account username would allow friends to easily find and follow each other to never miss out on posts.

### LINKED ISSUE #39: Back-end: Add Users to Database
#11 When users are registered, add them to the database.

### LINKED ISSUE #45: Front-End: Edit the Post Format
#10 Edit the Post UI to allow for Liking and Commenting (a 'Like' button and a text box with a submit button)

### LINKED ISSUE #46: Back-end: Create the Comment Form
#10 Create a form for the comment system.

### LINKED ISSUE #47: Database: Comment Entry
#10 When the submit button is pressed on the comment form, the comment is stored into the database.

### LINKED ISSUE #48: Front-end: Display all Comments
#10 All comments are to be gotten from the database and displayed on each post's page.

### LINKED ISSUE #49: Front-end: Display the Likes
#10 Display a counter that contains the number of likes for a given post.

### LINKED ISSUE #50: Front-end: Display Users who Liked a Post
#10 When the Like counter is clicked, show a text box containing all Users who Liked the post.

### LINKED ISSUE #51: Database: Like Entry
#10 When the Like button is pressed, the user is added into the database.

### LINKED ISSUE #55: Front-end: Non-Logged In UI
Create a nice home page for when the user is not logged in.

### LINKED ISSUE #60: Create page for user profile
Create a page that shows all of the user images as well as some information

### LINKED ISSUE #61: Create a post page
A post page that will show all information about the post, which includes likes, comments, user that posted.

### PR DESCRIPTION
**Pull request by alexbara2000**
Issues closed:
close #61 
close #60  
close #55 
close #51 
close #50 
close #49 
close #48 
close #47 
close #46  
close #45 
close #39 
close #11 
close #10   
issues worked on:
#25 
#23 
#9 
#56

### CODE REVIEW

**Review by MatteoGisondi — APPROVED** [review]
Good change to reduce user load, preperation for switch to database may need revision in next sprint.

**Review by jongjonathan — APPROVED** [review]
Good changes leading to users being able to access many features on a post with good formatting. New formatting design may be done in next sprint if needed with the upcoming changes.

File: app.py [hunk 1]
```diff
@@ -49,14 +54,18 @@ def index():
...
     try:
         username = str(session['_user_id'])
         print(username)
+        imageList = follower.getImagesToShow(username)
+        imagedict = {imageList[index*2+1]: imageList[index*2] for index in len(imageList)/2-5}
+        print(imagedict)
     except:
         print("An exception occurred")
         username = "Calasts53"
     print(username)
```
Comment (LiamBurchellReyes) [inline app.py:67]: This line of code likely isn't necessary except for testing.

File: follower.py [hunk 2]
```diff
@@ -94,27 +101,39 @@ def imagesForUser(user):
...
     # testing to make sure the user and followers are real
 
-    userIndex = followerlist[0].index(user)
-    followers = []
-    for x in range(len(followerlist[userIndex])):
-        if followerlist[userIndex][x] == "X":
-            followers.append(followerlist[x][0])
-            # checks the user to see all the people they follow and store it in a list
-    return followers
+    # userIndex = followerlist[0].index(user)
```
Comment (LiamBurchellReyes) [inline follower.py:117]: If this code isn't necessary, maybe delete it?

**Review by LiamBurchellReyes — APPROVED** [review]

**Pull request merged**